assessments/projects/tikv/self-assessment.md

```diff
@@ -0,0 +1,335 @@
+
+TiKV is inspired from Titanium (‘***Ti***’), as the creators kept the element’s property in mind. ‘***KV***’ stands for the ‘key-value’ combination in reference to the databases where it is primarily used. 
+
+### Background
+
+Key-value databases are optimized for fast data retrieval. They offer high performance for read/write operations, making them ideal for applications that require rapid access to data.
+
+TiKV is part of the Cloud Native Computing Foundation and created by PingCAP; it features ACID-compliant transactional APIs and ensures data consistency and high availability through the Raft Consensus Algorithm. The aim of the project is to provide a solution similar to Google Spanner and HBase, but with a focus on simplicity and ease of use for massive datasets.
+
+### Architecture
+
+<img src="./src/imgs/layer.png">
+
+* **Layers**
+  - OS Layer Storage Engine (RocksDB)
+  - Consensus Model
+  - Transaction (implements MVCC = multiversion concurrency control and distributed transactions)
+  - KV API
+  - Clients (using the gRPC protocol to connect to TiKV)
+
+* **Storage Engine**
+  - Uses [RocksDB](https://rocksdb.org/)
+    - High Read/Write Performance
+    - Easy to embed in Rust (TiKV is written in Rust)
+    - Stable
+  
+* **Consensus Model**
+  
+  <img src="./src/imgs/raft_consensus.png">
+  
+  * Raft Consensus Algorithm
+    - Client reads/writes to leader node
+    - Leader node replicates client requests to follower nodes
+    - The transaction is successful if the majority of the nodes (including leader and follower nodes) have fulfilled the client request
+    - If the leader node fails, then the up-to-date follower node becomes the leader
+      - There is a minimum of three nodes in a TiKV cluster
+    - As long as the majority of nodes are alive, the whole cluster is available
+  
+  * TiKV uses Range Sharding as opposed to Hash Sharding, which is used by other systems
+  
+    <img src="./src/imgs/sharding.png">
+  
+    * Range Sharding is better when scaling
+  
+  * TiKV brings together Replication (Raft Consensus Protocol) and Sharding (Range Sharding)
+  
+    <img src="./src/imgs/distributed_transaction.png">
+  
+  * When Scaling (i.e. adding extra nodes), the nodes are balanced and the leader of one node, becomes the leader of the new node
+  
+* **The Placement Driver**
+  - The Placement Driver Cluster stores “Region Information,” which the client can use to know which node to access
+  - The Placement Driver is “Highly Available”
+
+* **Client**
+  - The client uses the [gRPC](https://grpc.io/) protocol to interface with Placements Drivers and TiKV nodes
+
+<img src="./src/imgs/tikv_wholepic.png">
+
+### Actors
+
+* **Clients**
+* **TiKV Instance**
+  - Data Regions (within TiKV Nodes)
+* **Placement Driver**
+    - TimeStamp Oracle (within Placement Driver)
+
+### Actions
```

Comment:
Actions refer to what are the steps performed by each actor/multiple actors involved in this system, and the workflow including checks performed in achieving each of these capabilities. Refer https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md for a detailed description. 

Please update this section to reflect the same. A good example for this includes - https://github.com/cncf/tag-security/blob/main/assessments/projects/pixie/self-assessment.md#actions

Reply:
We have updated our actions section. We tried our best to follow the example you provided. Once more, please refer to our new pull request #1207. Thank you.